File: Semana 3 - Mini projetos/07 - Lista de tarefas no terminal/Lista_de_Tarefas_no_Terminal.js
// Description: taref list in terminal
const readline = require('readline');

function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    function menu(rl, tarefas) {
        console.log("\nMenu:");
        console.log("1. Adicionar tarefa");
        console.log("2. Listar tarefas");
        console.log("3. Sair");

        rl.question("Escolha uma opção: ", (escolha) => {
            if (escolha === "1") {
              // pede a tarefa e chama menu() de novo
              rl.question("Digite a tarefa: ", (tarefa) => {
                  tarefas.push(tarefa);
                    console.log("Tarefa adicionada!");
                    menu(rl, tarefas);               
                });
            } else if (escolha === "2") {
              // lista as tarefas e chama menu() de novo
                console.log("\nTarefas:");
                tarefas.forEach((tarefa, index) => {
                    console.log(`${index + 1}. ${tarefa}`);
                });
                menu(rl, tarefas);
            } else if (escolha === "3") {
                console.log("Saindo...");
                rl.close();
            } else {
                console.log("Opção inválida.");
                menu(rl, tarefas); // chama de novo
            }
        });
    }
    const tarefas = [];
    menu(rl, tarefas);
}

main();

File: Semana 3 - Mini projetos/07 - Lista de tarefas no terminal/Lista_de_Tarefas_no_Terminal.py
# Description: taref list in terminal
def main():

    tarefas = []
    while True:
        print("\nMenu:")
        print("1. Adicionar tarefa")
        print("2. Listar tarefas")
        print("3. Sair")
        escolha = input("Escolha uma opção: ")

        if escolha == "1":
            tarefa = input("Digite a tarefa: ")
            tarefas.append(tarefa)
            print("Tarefa adicionada!")

        elif escolha == "2":
            if not tarefas:
                print("Nenhuma tarefa encontrada.")
            else:
                print("Tarefas:")
                for idx, tarefa in enumerate(tarefas, start=1):
                    print(f"{idx}. {tarefa}")

        elif escolha == "3":
            print("Saindo...")
            break   
        else:
            print("Opção inválida. Tente novamente.")  


if __name__ == "__main__":
    main()